SalesforceCommerceCloud/pwa-kit
# W-16797377 [App Extensibility ⚙️] Create `pwa-kit-application-extensibility` project

# Description

This is a **HUGE** PR refactoring all our work and putting it in it's own project. Below is a list of changes that are included in this PR.

- Created new project and set up all the testing and typescript building etc.
- Removed previous solution of express extensibility and replaced it with same solution for react with the addition of a babel plugin to ensure that dev server works.
- Added a lot of tests for webpack and babel plugins
- Used handlebars for extensions loader.
- Updated extensions loader to use "loadable" for web targets.
- **Bonus:** Added hook/provider/context for getting the applied extensions. (This is used in the typescript minimal project homepage for now)
- many more...

# Note to Reviewers

Since this is a very large change, below is what I'm looking for feedback on. Given the feedback and required change, I may or may not create a ticket to address feedback.

1. General feedback on project organization.
2. Feedback on extension sdk interface with the rest of the projects, eg. using middleware to interface with the runtime and hoc to interface with the react app in the sdk
3. Review dependancies usage accross mono repo for this project. E.g. it's use in extensions
4. Test that you can generate working projects using extensibility and application extension projects and have the "dev" app work.
5. Review some of the webpack/babel plugin and loader tests

# Changes

- See above.

# How to Test-Drive This PR

- Run dev server, ensure that you are seeing extensions load as you normally would. E.g. check (/sample and /sample-page paths)
